zrml/court/src/lib.rs

```diff
@@ -957,6 +982,7 @@ mod pallet {
                 };
                 // sets round ends one after the other from now
                 court.update_round(round_timing);
+                new_round_ends = Some(court.round_ends.clone());
```

Comment:
Above you're using `let match`, but here you're using a mutable. Seems like inconsistent style.

Reply:
Also, why does `ne_round_ends` have to be defined inside the `if` expression?

Reply:
> Also, why does new_round_ends have to be defined inside the if expression?

It's defined outside by `let mut new_round_ends = None;`. In the case the condition `appeal_number < T::MaxAppeals::get() as usize` is true, then the round ends were actually updated by `court.update_round(round_timing);`. I don't want to emit an event where `new_round_ends` is written to be the old court round in the case of the last appeal (global dispute backer). So that's why I decided to use `None` for `new_round_ends` to indicate that this appeal belongs to the activation for global disputes.

Reply:
> Above you're using let match, but here you're using a mutable. Seems like inconsistent style.

I mean, true, it's another style than using `match`, and I know that you rather want to avoid unnecessary `mut` variables, but can you give me an example to achieve the same behaviour without using `mut` ?

Reply:
This would do it.

```rust
let (new_round_ends, ids_len_1) = if let ... {
    // ...
} else {
    (None, 0u32)
};
```

Not saying its better or worse, but I'd go with the same style everywhere.